### PR TITLE
DOC: Clarify difference between StringDtype(pyarrow) and ArrowDtype(string)

### PR DESCRIPTION
**Pull request by mroeschke**
Spawned from a discussion in #52156

### CODE REVIEW

File: doc/source/user_guide/pyarrow.rst [hunk 1]
```diff
@@ -35,6 +35,22 @@ which is similar to a NumPy array. To construct these from the main pandas data
...
+.. note::
+
+    The string alias ``"string[pyarrow]"`` maps to ``pd.StringDtype("pyarrow")`` which is not equivalent to
+    specifying ``dtype=pd.ArrowDtype(pa.string())``. Generally, operations on the data will behave similarly
+    except ``pd.StringDtype("pyarrow")`` can return NumPy-backed nullable types while ``pd.ArrowDtype(pa.string())``
+    will return :class:`ArrowDtype`.
+
+   .. ipython:: python
+
+      data = list("abc")
```
Comment (phofl) [inline doc/source/user_guide/pyarrow.rst:47]: Needs a pyarrow import

File: doc/source/user_guide/pyarrow.rst [hunk 2]
```diff
@@ -115,7 +131,7 @@ The following are just some examples of operations that are accelerated by nativ
    ser.isna()
    ser.fillna(0)
 
-   ser_str = pd.Series(["a", "b", None], dtype="string[pyarrow]")
+   ser_str = pd.Series(["a", "b", None], dtype=pd.ArrowDtype(pa.string()))
```
Comment (phofl) [inline doc/source/user_guide/pyarrow.rst:134]: same here probably

**Review by phofl — COMMENTED** [review]
wording looks good

**Comment by jbrockmendel** [comment]
Is there a plan to deprecate the older one and if so should that be mentioned here?

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @mroeschke 

There is a discussion about deprecating this somewhere but no consensus yet.

**Comment by mroeschke** [comment]
Yeah the discussion started at https://github.com/pandas-dev/pandas/issues/48469

**Pull request merged**